src/v3/src/transformer/oktaVerify/transformOktaVerifyEnrollPoll.ts

```diff
@@ -127,7 +184,155 @@ export const transformOktaVerifyEnrollPoll: IdxStepTransformer = ({ transaction,
+            ),
+          },
+        } as DescriptionElement,
+        {
+          type: 'Description',
+          noMargin: true,
+          options: {
+            content: loc(
+              'oie.enroll.okta_verify.setup.customUri.noPrompt',
+              'login',
+              undefined,
+              { $1: { element: 'span', attributes: { class: 'strong' } } },
+            ),
+          },
+        } as DescriptionElement,
+        {
+          type: 'Description',
+          noMargin: true,
+          options: {
+            content: loc('oie.enroll.okta_verify.setup.customUri.makeSureHaveOV', 'login'),
+          },
+        } as DescriptionElement,
+      );
+    } else {
+      sameDeviceOVElements.push(
+        {
+          type: 'Description',
+          noMargin: true,
+          options: {
+            content: loc(
+              'oie.enroll.okta_verify.setup.customUri.makeSureHaveOVToContinue', 'login',
+            ),
+          },
+        } as DescriptionElement,
+      );
+    }
+    if (deviceMap.platformLC) {
+      let platformSupported = true;
+      let svgIcon;
+      switch (deviceMap.platformLC) {
+        case 'android':
+          svgIcon = GoogleStoreIcon;
+          break;
+
+        case 'ios':
+          svgIcon = AppleStoreIcon;
+          break;
+
+        case 'windows':
+          svgIcon = WindowsStoreIcon;
+          break;
+
+        case 'osx':
+          svgIcon = OSXStoreIcon;
+          break;
+
+        default:
+          platformSupported = false;
+      }
+      if (platformSupported) {
+        sameDeviceOVElements.push(
+          {
+            type: 'ImageLink',
+            options: {
+              href: deviceMap.downloadHref,
+              altText: loc('enroll.oda.step3', 'login'),
+              alignment: 'center',
+              svgIcon,
+              dataSe: 'app-store-link',
+              marginBlockStart: '20px',
+            },
+          } as ImageLinkElement,
+        );
+      }
+    }
+    sameDeviceOVElements.push(
+      {
+        type: 'Description',
+        noMargin: true,
+        options: {
+          content: loc(
+            'oie.enroll.okta_verify.setup.customUri.setup',
+            'login',
+            undefined,
+            { $1: { element: 'span', attributes: { class: 'strong' } } },
+          ),
+        },
+      } as DescriptionElement,
+    );
+
+    if (deviceMap.setupOVUrl) {
+      sameDeviceOVElements.push(
+        {
+          type: 'Button',
+          label: loc('oie.enroll.okta_verify.setup.title', 'login'),
+          options: {
+            step: '',
+            type: ButtonType.BUTTON,
+            variant: 'primary',
+            onClick: () => {
+              if (isAndroid()) {
+                Util.redirectWithFormGet(deviceMap.setupOVUrl);
+              } else {
+                // @ts-ignore
+                window.location.assign(deviceMap.setupOVUrl);
```

Comment:
```suggestion
                window.location.assign(deviceMap.setupOVUrl!);
```
It looks like you're using `@ts-ignore` since you know that `setupOVUrl` must be defined at this point in the code. You can use [TypeScript's non-null assertion operator](https://www.typescriptlang.org/docs/handbook/2/everyday-types.html#non-null-assertion-operator-postfix-) to tell TS that you know this value is actually defined.

Reply:
This is interesting, I would have expected TS to realize that `deviceMap.setupOVUrl` is set to a non-null/undefined value due to the if condition `if (deviceMap.setupOVUrl) {` just above. @vakaevich-okta Are you sure thing is needed? If you remove it do you still get a TS error?

Reply:
If you still get an error, using the `!` operator should be ok in this instance because we know for a fact it is set.

Reply:
please see https://github.com/okta/okta-signin-widget/pull/3526/commits/eef180e94da1dfc5f227044b14a1c493e9270d55